packages/type-utils/src/typeOrValueSpecifiers/typeDeclaredInPackageDeclarationFile.ts

```diff
@@ -30,15 +32,36 @@ function typeDeclaredInDeclarationFile(
   declarationFiles: ts.SourceFile[],
   program: ts.Program,
 ): boolean {
-  // Handle scoped packages: if the name starts with @, remove it and replace / with __
-  const typesPackageName = packageName.replace(/^@([^/]+)\//, '$1__');
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#escaping
+  const escapeAlternates = (alternates: string[]): string =>
+    alternates
+      .map(alternate => alternate.replaceAll(/[.*+?^${}()|[\]\\]/g, '\\$&'))
+      .join('|');
 
-  const matcher = new RegExp(`${packageName}|${typesPackageName}`);
+  const packageNameRegExp = escapeAlternates([
+    packageName,
+    // Handle scoped packages: if the name starts with @, remove it and replace / with __
+    `@types/${packageName.replace(/^@([^/]+)\//, '$1__')}`,
```

Comment:
[Bug] `@types/` should not be hardcoded to always be considered. From https://www.typescriptlang.org/tsconfig/#typeRoots:

> If `typeRoots` is specified, _only_ packages under `typeRoots` will be included.